src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -301,9 +302,22 @@ public void Complete()
                 _buf.EndCopyMode();
 
                 _connector.SendMessage(CopyDoneMessage.Instance);
-                Expect<CommandCompleteMessage>(_connector.ReadMessage());
+
+                var msg1 = _connector.ReadMessage();
+                Expect<CommandCompleteMessage>(msg1);
                 Expect<ReadyForQueryMessage>(_connector.ReadMessage());
                 _state = ImporterState.Committed;
+
+                CommandCompleteMessage cmdComplete;
+                switch (msg1.Code)
+                {
+                    case BackendMessageCode.CompletedResponse:
+                        cmdComplete = (CommandCompleteMessage)msg1;
+                        if (cmdComplete.StatementType == StatementType.Copy)
```

Comment:
Once again, how can the statement type be anything other than copy? You can put this in a `Debug.Assert()` but it doesn't seem to make sense to test for it...

Reply:
c) I have been living in F# world recently, which is far more explicit using its type system to disallow invalid states. So my caution in jumping into C# without knowing the codebase probably led me be ok in over-using available checks. 

For example: as a completely new user to this codebase and after your prompting, it is (now) intuitive that the following is true:

If msg1 is of type CommandCompleteMessage, then msg1.Code must equal BackendMessageCode.CompletedResponse.  

However, when I was adding this code, at first glance I could not be certain of that relationship even though it seemed correct, and so out of caution and limited time, I re-used a check from another part of the codebase. 
In F#, I would at this point briefly ponder whether to restructure the types/classes in such a way to remove (not require) these checks, a common refactoring, but this has been less natural to do in C# until the recent introduction of discriminated unions.  

Your comment is well noted, and I will revise and streamline this series of logic. 
I agree, the checks are unnecessary now that I understand the type/property relationships.